opal/corelib/array.rb

```diff
@@ -544,14 +561,14 @@ def clear
   end
 
   def clone
-    copy = []
+    copy = self.class.allocate
     copy.copy_singleton_methods(self)
     copy.initialize_clone(self)
     copy
   end
 
   def dup
-    copy = []
+    copy = self.class.allocate
     copy.initialize_dup(self)
     copy
   end
```

Comment:
`#clone` and `#dup` can probably be removed to:

https://github.com/opal/opal/blob/a8669f9fd1433f192564ee4a2a43b04b39dd814e/opal/corelib/kernel.rb#L138-L163